queue/server.go

```diff
@@ -0,0 +1,121 @@
+package queue
+
+import (
+	"sync"
+
+	"context"
+	"errors"
+	"time"
+
+	"github.com/centrifuge/gocelery"
+	logging "github.com/ipfs/go-log"
+)
+
+const (
+	BlockHeightParam string = "BlockHeight"
+	TimeoutParam     string = "Timeout"
+)
+
+var log = logging.Logger("queue-server")
+
+// Config is an interface for queue specific configurations
+type Config interface {
+
+	// GetNumWorkers gets the number of background workers to initiate
+	GetNumWorkers() int
+
+	// GetWorkerWaitTime gets the worker wait time for a task to be available while polling
+	// increasing this may slow down task execution while reducing it may consume a lot of CPU cycles
+	GetWorkerWaitTimeMS() int
+}
+
+// TaskType is a task to be queued in the centrifuge node to be completed asynchronously
+type TaskType interface {
+
+	// TaskTypeName of the task
+	TaskTypeName() string
+}
+
+// TaskResult represents a result from a queued task execution
+type TaskResult interface {
+
+	// Get the result within a timeout from the queue task execution
+	Get(timeout time.Duration) (interface{}, error)
+}
+
+// Server represents the queue server currently implemented based on gocelery
+type Server struct {
```

Comment:
I think this can be unexported and a constructor could return the Server interface instead